### DISCUSSION TITLE
Add AWS Cost Explorer API

### DISCUSSION THREAD

**Discussion by Issif** [grafana/grafana#41812] [Data Sources]
Hi,

AWS has launched yesterday a new API to access all raw data used by their Cost Explorer. It will be amazing to be able to create custom dashboard in Grafana to follow these new metrics. I know CloudWatch datasource is already able to display metrics about Costs, but Cost Explorer data are much more detailed.

Regards,

**Comment by ghost** [comment]
+1

**Comment by mtanda** [comment]
Just note. This is not CloudWatch feature. It is another AWS service.
https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/cost-explorer-api.html

To support this, need to add datasource plugin.

**Comment by PaulW** [comment]
+1 for this, as cloudwatch doesn't play nice with billing metrics on linked accounts.

**Comment by moshikoy** [comment]
+1 for this

**Comment by kcl11** [comment]
+1

**Comment by Sher-Chowdhury** [comment]
+1

**Comment by jcmartins** [comment]
+1

**Comment by ldecaro** [comment]
+1

**Comment by chamilbmll** [comment]
+1

**Comment by mateusrangel** [comment]
+1

**Comment by alanwds** [comment]
+1

**Comment by alanwds** [comment]
I just wrote a prometheus exporter to use CE API. I hope that be usefull:

https://github.com/alanwds/aws-cost-exporter

**Comment by suraganijanakiram** [comment]
+1 

This is a great feature to have since now a days every one is moving to cloud and AWS is having the majority of the share in the cloud.

**Comment by balda** [comment]
Be aware of the cost of this API: $0.01 per request!

**Comment by suraganijanakiram** [comment]
> Be aware of the cost of this API: $0.01 per request!

i think if they can put an option to send request like below then i hope cost wont be a big problem.
@hourly
@daily
@weekly
@monthly
etc.

**Comment by hmfritz** [comment]
+1 I'd like the ability to filter by tag for AWS Costs

**Comment by JamesDowning** [comment]
Is there any new information on this? The AWS/Billing data is very basic.

**Comment by aryak007** [comment]
Is there an update on creating a data source which talks to the AWS CE API? A timeline, perhaps?

**Comment by dimisjim** [comment]
Hey @vickyyyyyyy 

Could you elaborate on the reasoning behind moving this "issue" to a "discussion"?
Or maybe, could you provide a few pointers to some of us (people following this and hoping for quite some time now for this to be introduced to Grafana) on how we could help this discussion, moving forward?

Thanks in advance!

**Reply by vickyyyyyyy** [reply]
hey @dimisjim!

We are moving from issues to discussions to try to improve the way we handle and organise feature requests (please see the [forum](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341) for more information).

This feature request is something we are aware has been asked for, for some time now but we've not been able to add it to the roadmap in the near future yet. We will let you know if there are any updates and keep this discussion updated on any new development

**Comment by ubershloder** [comment]
+1

**Comment by vickyyyyyyy** [comment]
in lieu of having a specific plugin that uses the cost explorer API, the way we are currently suggesting users to do this is with our Athena plugin:
https://grafana.com/grafana/plugins/grafana-athena-datasource/ https://docs.aws.amazon.com/cur/latest/userguide/cur-query-athena.html

would this be sufficient for your use cases?

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This discussion was converted into an issue, which is available [here](https://github.com/grafana/grafana/issues/73444).
